pkg/upgrade/upgrade.go

```diff
@@ -78,29 +80,62 @@ func ManagedInstances(ctx context.Context, c client.Client, reader client.Reader
 }
 
 // ManagedInstance performs the necessary changes to bring the given Jaeger instance to the current version
-func ManagedInstance(ctx context.Context, client client.Client, jaeger v1.Jaeger) (v1.Jaeger, error) {
+func ManagedInstance(ctx context.Context, client client.Client, jaeger v1.Jaeger, latestVersion string) (v1.Jaeger, error) {
 	tracer := global.TraceProvider().GetTracer(v1.ReconciliationTracer)
 	ctx, span := tracer.Start(ctx, "ManagedInstance")
 	defer span.End()
 
-	if v, ok := versions[jaeger.Status.Version]; ok {
+	currentSemVersion, err := semver.NewVersion(jaeger.Status.Version)
+
+	if err != nil {
+		jaeger.Logger().WithFields(log.Fields{
+			"instance":  jaeger.Name,
+			"namespace": jaeger.Namespace,
+			"to":        latestVersion,
+		}).WithError(err).Warn("Skipped for update")
```

Comment:
This error needs to be more specific - i.e. change the "to" field value to `jaeger.Status.Version` and the warning message to "Failed to parse current Jaeger instance version. Unable to perform upgrade".

Reply:
Maybe need to return, otherwise wouldn't that cause problems further down?